### PR TITLE
Add must_exist to PerfGraphData and various PerfGraph cleanup

### LINKED ISSUE #19353: Allow retreival of sections in `PerfGraphData` that have not ran yet
## Reason
Currently, if you have a postprocessor like this:

```
[Postprocessors/calls]
  type = PerfGraphData
  data_type = CALLS
  section_name = "FEProblem::computeJacobianInternal"
  execute_on = TIMESTEP_END
[]
```

and residuals are not executed until many timesteps into the run, it will error with something like:

```
Unknown section_name: "FEProblem::computeResidualInternal"
```

This is not ideal. We need to support this capability.

## Design
Add a `must_exist` parameter to `PerfGraphData` that allows the error to be skipped if `must_exist = true`.

## Impact
Increased `PerfGraph` functionality.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #19353

### CODE REVIEW

**Comment by loganharbour** [comment]
FYI @gambka @sapitts @singhgp4321

File: framework/src/postprocessors/PerfGraphData.C [hunk 1]
```diff
@@ -20,44 +17,27 @@ InputParameters
...
+      "Retrieves performance information about a section from the PerfGraph.");
 
   params.addRequiredParam<std::string>("section_name", "The name of the section to get data for");
-
   params.addRequiredParam<MooseEnum>(
-      "data_type", data_type, "The type of data to retrieve for the section_name");
-
-  params.addClassDescription("Retrieves timing information from the PerfGraph.");
+      "data_type", PerfGraph::dataTypeEnum(), "The type of data to retrieve for the section_name");
+  params.addParam<bool>("must_exist", true, "foo");
```
Comment (friedmud) [inline framework/src/postprocessors/PerfGraphData.C:26]: I'm not sure that `foo` is the best documentation for this!

**Review by friedmud — CHANGES_REQUESTED** [review]
A couple of small comments.

File: framework/src/postprocessors/PerfGraphData.C (hunk 1 above)
Comment (loganharbour) [reply framework/src/postprocessors/PerfGraphData.C:26]: oops

**Review by friedmud — APPROVED** [review]
👍

**Review by zachmprince — APPROVED** [review]
Looks good to me

**Pull request merged**